samples/subsys/fs/nffs/src/main.c

```diff
@@ -0,0 +1,149 @@
+#include <stdio.h>
+#include <stddef.h>
+#include <string.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <errno.h>
+#include <misc/printk.h>
+#include <misc/byteorder.h>
+#include <fs.h>
+#include <nrf.h>
+#include <device.h>
+#include <gpio.h>
+#include <board.h>
+
+int lock=0;
+
+struct device *button_device[1];
+
+static struct k_work button_work;
+
+static void button_pressed(struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	k_work_submit(&button_work);
+}
+
+void press(struct k_work *work)
+{
+	char buf[20];
+	int err=0;
+
+	fs_file_t fp;
+
+	err = fs_open(&fp,"/0.txt");
+	if (err != 0) {
+		printk("error %d while opening for read\n\r", err);
+	}
+
+	err = fs_read(&fp, buf, sizeof(buf));
+	if (err < 0) {
+		printk("error %d while reading\n\r", err);
+	}
+
+	printk("Data read from file is --> %s \n", buf);
+
+	err = fs_close(&fp);
+	if (err != 0) {
+		printk("error %d while closing after read\n\r", err);
+	}
+}
+
+static int test(int idx)
+{
+	int err;
+	char name[MAX_FILE_NAME + 1];
+	fs_file_t fp;
+	static const char msg[] = "Hello World\0";
+	char buf[sizeof(msg)];
+
+	/* Note that NFFS needs an absolute path */
+	sprintf(name, "/%d.txt", idx);
+
+	/* Delete the old file, if any */
+	fs_unlink(name);
+
+	/* Open and write */
+	err = fs_open(&fp, name);
+	if (err != 0) {
+		printk("%s: error %d while opening for write\n", name, err);
+		return err;
+	}
+
+	err = fs_write(&fp, msg, sizeof(msg));
+	if (err < 0) {
+		printk("%s: error %d while writing\n", name, err);
+		return err;
+	}
+
+	if (err != sizeof(msg)) {
+		printk("%s: tried to write %d bytes, only wrote %d\n", name,
+		       sizeof(msg), err);
+		return err;
+	}
+
+	/* Close the file and read the data back */
+	err = fs_close(&fp);
+	if (err != 0) {
+		printk("%s: error %d while closing after write\n", name, err);
+		return err;
+	}
+
+	struct fs_dirent entry;
+
+	if(fs_stat(name,&entry)==0)
```

Comment:
Why is this being wrapped in a call to fs_stat()?

If it's just trying to check whether or not the file exists, doesn't the call to fs_open() a few lines below handle that?